02-peer-protocol.md

```diff
@@ -390,6 +400,377 @@ would create a Denial of Service risk; therefore, forgetting it is recommended
+```
+
+`output_info`.`script` is the locking script for the output.
+
+The channel funding output is not exchanged, as it can be derived
+independently.
+
+#### The `commitment_signed2` Message
+
+This message exchanges the counterparty's signature for the
+first commitment transaction, so it can broadcast the funding
+transaction knowing that the funds can be redeemed.
+
+1. type: 59 (`commitment_signed2`)
+2. data:
+  See [`commitment_signed`](#commiting-updates-so-far-commitment_signed).
+
+#### Requirements:
+
+The sending node:
+  - MUST derive the `channel_id` from the SHA256(
+    `open_channel2`.`revocation_basepoint`|`accept_channel2`.`revocation_`
+     `basepoint`)
+  - MUST set signature to the valid signature, using its `funding_pubkey`
+    for the initial commitment transaction, as defined in [BOLT #3](03-transactions.md#per-commitment-secret-requirements).
+  - MUST set `num_htlcs` to zero.
+
+A receiving node:
+  - if the `num_htlcs` is not zero:
+    - MUST fail the channel.
+  - if sent in response to a `funding_compose` message:
+    - MUST respond with a `commitment_signed` message.
+  - otherwise:
+     - MUST respond with a `funding_signed2` message.
+
+#### Rationale
+There are no HTLC's yet, so the number must be zero. FIXME: use this
+for `will_fund_for_food` bonds?
+
+Note that the channel id for `open_channel2` is fixed as the SHA256 of the
+concatenated revocation points, and will not update. This allows stable
+channel tracking across RBFs and splicing attempts.
+
+### The `funding_signed2` Message
+
+This message exchanges the witness data for the inputs that were
+originally sent in the `funding_compose` message.
+
+1. type: 60 (`funding_signed2`)
+2. data:
+  [`32`:`channel_id`]
+  [`2`:`num_witnesses`]
+  [`num_witnesses`\*`witness_stack`]
+
+1. subtype: `witness_stack`
+2. data:
+  * [`2`:`num_input_witness`]
+  * [`num_input_witness`\*`witness_element`]
+
+1. subsubtype: `witness_element`
+2. data:
+  * [`2`:`len`]
+  * [`len`:`witness`]
+
+#### Requirements
+The sending node:
+  - MUST set `witness` to the marshalled witness data for each of its
+    inputs, in funding transaction order. FIXME: add reference
+  - MUST remember the details of this funding transaction.
+  - MUST NOT send a `witness_stack` whose length exceeds the
+    corresponding `max_witness_len`.
+  - if they have NOT received a valid `commitment_signed` message:
+    - MUST NOT send a `funding_signed2` message.
+
+The receiving node:
+  - if the `witness_stack` length exceeds the corresponding `max_witness_len`:
+    - MUST error.
+  - if the recipient was the sender of `open_channel2`:
+    - SHOULD reply with its own `funding_signed2`.
+  - SHOULD apply `witness`es to the funding transaction and
+    broadcast the result.
+
+#### Rationale
+Exchanging witness data allows both sides to broadcast the funding
+transaction and maintains information symmetry.
+
+### The `funding_locked2` Message
```

Comment:
Again, prefer direct 'funding_locked' reuse.

Reply:
inclusion of the `funding_tx_id` means we need a v2 for this